Movecraft/src/main/java/net/countercraft/movecraft/processing/tasks/detection/DetectionTask.java

```diff
@@ -339,35 +352,79 @@ private DetectAction(ConcurrentLinkedQueue<MovecraftLocation> currentFrontier, C
         @Override
         public void run() {
             MovecraftLocation probe;
+
             while((probe = currentFrontier.poll()) != null) {
-                visitedMaterials.computeIfAbsent(movecraftWorld.getMaterial(probe), Functions.forSupplier(ConcurrentLinkedDeque::new)).add(probe);
+                BlockData bd = movecraftWorld.getData(probe);
+                Material m = bd.getMaterial();
```

Comment:
Add a `CraftType` entry to allowlist this process based on materials, and another one to blanket enable the feature.

Reply:
Just to be clear i need to create 2 entries like this?
directionDependency: true
directionDependentMaterials:
 - ladder

Reply:
no, the bool one should just be something to make it so that all materials are tested by default

Reply:
That would be insanity, I tried that and even if they are not facing a block inside a craft won't be put inside, the only way I could spot that was thanks to some robust logging that players don't have

Reply:
Sorry, I meant all reasonable materials - we can have a base list somewhere

Reply:
I would say to only use the `directionDependentMaterials` option, and if it is not used, just used a small list of the vary basic stuff like ladder and wall signs.

Reply:
Perhaps it would make sense to have `directionDependency` default to everything, but if specified will become a whitelist (or blacklist)?  That way we wouldn't need two separate flags.

Reply:
I think that is what I am doing for now, there is a default whitelist with all reasonable materials (note: we can't do everything as I said because it would be insanity and the staff would be forced to override the flag and set it to panic as some crafts would just stop working, all directionals include furnaces chests and more which is problematic), and in case it is overriden player can specify a whitelist. 

Implementing a blacklist or using all materials here would create chaos in my opinion, imagine joining a movecraft server with this default to "everything" and you can't pilot any craft because you furnace/chest's opposite facing block is air